src/storage/src/source/postgres.rs

```diff
@@ -286,6 +286,8 @@ pub enum DefiniteError {
     MissingColumn,
     #[error("failed to parse COPY protocol")]
     InvalidCopyInput,
+    #[error("The replication timeline id has changed. Expected {0} but got {1}")]
```

Comment:
the "style" of the other messages here is a good pattern to follow, e.g. maybe something like "invalid timeline ID from PostgreSQL server"